pixeldata/src/attribute.rs

```diff
@@ -152,20 +153,90 @@ pub fn pixel_data<D: DataDictionary + Clone>(
         .context(MissingRequiredSnafu { name })
 }
 
+
+// todo: maybe generalize in the future
+fn get_f64_from_per_frame<D: DataDictionary + Clone>(obj: &FileDicomObject<InMemDicomObject<D>>, selector: [Tag; 2], default: f64) -> Option<Vec<f64>>{
```

Comment:
I'd like to generalize these methods to return a `Vec` of `DataElement`s, and then let the upstream method (`rescale_slope`/`rescale_intercept`/`voi_lut_function`) handle type conversions and errors, but I'm really struggling with figuring out how to use snafu `context`  in these chains